nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractBulkComputation.java

```diff
@@ -74,18 +76,29 @@ public AbstractBulkComputation(String name, int nbOutputStreams) {
     public void processRecord(ComputationContext context, String inputStreamName, Record record) {
         BulkBucket bucket = BulkCodecs.getBucketCodec().decode(record.getData());
         command = getCommand(bucket.getCommandId());
-        if (command == null) {
-            // this requires a manual intervention, the kv store might have been lost
-            getLog().error(String.format("Stopping processing, unknown command: %s, offset: %s, record: %s.",
-                    bucket.getCommandId(), context.getLastOffset(), record));
-            context.askForTermination();
-            return;
-        }
-        for (List<String> batch : Lists.partition(bucket.getIds(), command.getBatchSize())) {
-            processBatchOfDocuments(batch);
+        if (command != null) {
+            for (List<String> batch : Lists.partition(bucket.getIds(), command.getBatchSize())) {
+                processBatchOfDocuments(batch);
+            }
+            endBucket(context, bucket.getIds().size());
+            context.askForCheckpoint();
+        } else {
+            if (isAbortedCommand(bucket.getCommandId())) {
```

Comment:
don't we want to be able to abort a running command ?

Reply:
ok new version uses a cache of 1min for commands to reduce kv store roundtrip, the abortion is then delayed up to 1min but this should be ok because the purpose is to cancel very long bulk or broken action (where the delay does not apply)